### PR TITLE
Separate module for lifting infrastructure

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

File: autogen/build.rs [hunk 1]
```diff
@@ -192,7 +192,7 @@ fn main() {
         instructions.ops,
     );
     write_formatted(
-        &autogen_src_dir.join("../rspirv/sr/autogen_module.rs"),
-        instructions.module_logic,
+        &autogen_src_dir.join("../rspirv/lift/autogen_context.rs"),
```
Comment (antiagainst) [inline autogen/build.rs]: Shouldn't we call this as `autogen_lift_context.rs` to be clear then?

Comment (kvark) [reply autogen/build.rs]: I had it called this for some time but then figured that "lift" is already in the path, so there is no need to repeat ourselves here. I don't mind either way :)
@AIOOB - do you have a preference?

File: autogen/sr.rs [hunk 2]
```diff
@@ -104,14 +104,14 @@ impl OperandTokens {
             ),
             "PairLiteralIntegerIdRef" => (
                 //TODO: Token<BasicBlock>
```
[the remark this answers is not in the document]
Comment (kvark) [reply autogen/sr.rs]: indeed!

**Comment by kvark** [comment]
Ok, I'm open to revisiting this decision. Currently, the autogen files sometimes filed as the types, sometimes not (when there are many types in a file).

**Comment by kvark** [comment]
Oh, I didn't push the fix to remove the TODO. Sorry about that!

**Pull request merged**